xbmc/interfaces/json-rpc/PlayerOperations.cpp

```diff
@@ -153,32 +153,38 @@ JSONRPC_STATUS CPlayerOperations::GetItem(const std::string &method, ITransportL
       }
       else if (player == Video)
       {
-        if (!CVideoLibrary::FillFileItem(g_application.CurrentFile(), fileItem, parameterObject))
+        if (!CVideoLibrary::FillFileItem(fileItem->GetPath(), fileItem, parameterObject))
         {
+          // Fallback to item details held by GUI but preserve path
+          //! @todo  check this fallback still needed, can GUI item have more details than app item?
           const CVideoInfoTag *currentVideoTag = CServiceBroker::GetGUI()->GetInfoManager().GetCurrentMovieTag();
           if (currentVideoTag != NULL)
           {
             std::string originalLabel = fileItem->GetLabel();
+            std::string originalPath = fileItem->GetPath();
             fileItem->SetFromVideoInfoTag(*currentVideoTag);
             if (fileItem->GetLabel().empty())
               fileItem->SetLabel(originalLabel);
+            fileItem->SetPath(originalPath);  // Ensure path unchanged
```

Comment:
wrong. see comment below.

Reply:
See comment above - SetVideoInfoTag possibly changes path

Reply:
> SetVideoInfoTag possibly changes path

please point me to the code where it does

Reply:
`SetVideoInfoTag` sets path https://github.com/xbmc/xbmc/blob/067538b50084397f021ecea02e52f0d3825adbba/xbmc/FileItem.cpp#L1650

Reply:
check out this comment: https://github.com/xbmc/xbmc/blob/067538b50084397f021ecea02e52f0d3825adbba/xbmc/FileItem.cpp#L1736

changing path after construction is not desired. Instead of calling SetVideoInfoTag, clients can construct a new object.

VideoInfoTag is stored in db with path as primary key. If path of VideoInfoTag does not equal originalPath things really go wrong. 

If Tag obtained from gui is different to the one already got from CApp, things go wrong too. If you ever intend to improve the API by eliminating unneeded/wrong code, I suggest to place some comments and log warnings. 

Apart from this, the changes here do not change behaviour (just fixes segfault). I was referring to code/design.

Reply:
Understood. I will extend code comments